shap/explainers/_tree.py

```diff
@@ -355,7 +355,12 @@ def shap_values(self, X, y=None, tree_limit=None, approximate=False, check_addit
             if self.model.model_type == "xgboost":
                 import xgboost
                 if not isinstance(X, xgboost.core.DMatrix):
-                    X = xgboost.DMatrix(X)
+                    # Retrieve any DMatrix properties if they have been set on the class
+                    dmatrix_props = [p for p in dir(self) if p.startswith("dmatrix_")]
```

Comment:
With the suggested change below we have
```python
dmatrix_props = getattr(self, 'dmatrix_props', {})
```
And can just pass this as kwargs to the DMatrix init